packages/filesystem/src/node/encoding.ts

```diff
@@ -0,0 +1,51 @@
+/********************************************************************************
+ * Written by Xuye Cai <a1994846931931@gmail.com>, June 2019
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+// copied from vscode: https://github.com/Microsoft/vscode/blob/master/src/vs/base/node/encoding.ts
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+const jschardet = require('jschardet');
+const MINIMUM_THRESHOLD = 0.2;
+jschardet.Constants.MINIMUM_THRESHOLD = MINIMUM_THRESHOLD;
+const IGNORE_ENCODINGS = ['ascii', 'utf-8', 'utf-16', 'utf-32'];
+export async function guessEncodingByBuffer(buffer: Buffer): Promise<string | undefined> {
```

Comment:
please put them in DI context, see https://github.com/theia-ide/theia/wiki/Coding-Guidelines#di-function-export